src/renderers/dom/client/MarkupMismatchError.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule MarkupMismatchError
+ */
+
+'use strict';
+
+var ELEMENT_NODE_TYPE = 1;
+var TEXT_NODE_TYPE = 3;
+var COMMENT_NODE_TYPE = 8;
+
+// we cannot use class MarkupMismatchError extends Error {} because Babel cannot extend built-in
+// objects in a way that allows for instanceof checks.
+// this error subclassing code is copied and modified from
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types
+// According to https://developer.mozilla.org/en-US/docs/MDN/About#Copyrights_and_licenses, all code in MDN
+// added on or after August 20, 2010 is public domain, and according to the wiki page history, the earliest
+// version of this code was added on June 7, 2011:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error$revision/62425
+function MarkupMismatchError(message, node, serverVersion, clientVersion) {
+  this.name = 'MarkupMismatchError';
+  this.message = message || 'The pre-rendered markup did not match the component being rendered.';
+  this.node = node;
+  this.serverVersion = serverVersion;
+  this.clientVersion = clientVersion;
+  this.stack = (new Error()).stack;
```

Comment:
Reading further on this at [this MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack) and [this MSDN article](https://msdn.microsoft.com/en-us/library/windows/apps/hh699850.aspx), it seems to me that this will grab a stack for all browsers where a stack is available. Stack is completely non-standard, so I can't guarantee that it will work absolutely everywhere. 

It should be fine on platforms where there is no stack property.

Note also that every place in the current code where MismatchError is thrown, it is caught without looking at the stack.

Given all that, I'm going to leave the code as is. Let me know if you disagree, and thanks for the comment!